Bug problem with asset register

lomamech opened the pull request:
- Hide the asset where quantity in stock is 0
- Add HAVING filters in SQL Clause format
**_Bonus_**
- Set Serial Number not required in lot entry stock
- Dont display expiry information for non Expirable asset

closes https://github.com/IMA-WorldHealth/bhima/issues/6842

lomamech committed:
bug(Asset Register)
- Hide the asset where quantity in stock is 0
- Add HAVING filters in SQL Clause format

closes #6842

lomamech committed:
Set Serial Number not required in lot entry stock

lomamech committed:
Dont display expiry information for non Expirable asset

lomamech committed:
Test server:unit fix

lomamech committed:
Test: Verification of tests for semaphore

lomamech committed:
Test: restaure integration test

jmcameron commented on server/controllers/stock/core.js:
```diff
@@ -318,6 +319,8 @@ async function getAssets(params) {
   }
   filters.setGroup(groupByClause);
 
+  filters.setHaving(havingClause);
```
👍

jmcameron commented on server/lib/filter.js:
```diff
@@ -215,8 +228,9 @@ class FilterParser {
+    const having = this._having;
 
-    return `${sql} WHERE ${conditionStatements} ${group} ${order} ${limitCondition}`;
+    return `${sql} WHERE ${conditionStatements} ${group} ${having}${order} ${limitCondition}`;
```
I think you may want to add a space between the ${having} and ${order} so the developer does not need to add a space when using both.

jmcameron reviewed (commented):
I looked over the code and pointed out a minor issue.

I tested (with bhima_test) and it all seems work work fine.

However, when I looked at the articles in stock and hovered over the warning triangle for Quinine, I noticed that the tooltip had not been fully translated.  See:
![image](https://user-images.githubusercontent.com/62145/196482250-06e1fb54-e1df-42f8-a8c2-c58b08c1a2f3.png)

This can be fixed by editing client/src/modules/stock/inventories/registry.js and adding a line to translate the unit_type after line 275 so it looks like this:

        // set status flags
        rows.forEach(row => {
          setStatusFlag(row);
          row.status_translated = $translate.instant(Stock.statusLabelMap(row.status));
          row.unit_type = $translate.instant(row.unit_type);
        });

lomamech committed:
improvement
- Translate unit type
- Sanitaze filter prevent blank character when formatting SQL queries in
  cas the HAVING clause is not needed
